packages/cozy-sharing/src/helpers/files.js

```diff
@@ -8,7 +8,11 @@ export const fetchFilesPaths = async (client, doctype, files) => {
   const filePaths = files
     .map(f => parentDirs.data.find(d => d.id === f.dir_id))
     .filter(parentDir => parentDir !== undefined)
-    .map(parentDir => {
+    .map((parentDir, index) => {
+      // If the file is a directory, we return its path directly
+      if (files[index].type === 'directory') {
```

Comment:
I'm not sure how well it works with an index, because since we're doing a filter before the size of the files array and the one we're mapping here is no longer the same 🤔

Reply:
> I'm not sure how well it works with an index, because since we're doing a filter before the size of the files array and the one we're mapping here is no longer the same 🤔

that's true, should take the index before any filtering. Though I'm unsure in which scenarios the files array is longer than 1, I couldn't find such a case

Reply:
I think it's possible to have several here: https://github.com/cozy/cozy-libs/blob/e3e6b6ca27bfcc2c61893eeb763c43f926308630/packages/cozy-sharing/src/SharingProvider.jsx#L203 

You could do a unit test to check that this case works and adapt the function to handle it

Reply:
Updated with more streamlined logic. If any directory in input (should always be one according to code, but never sure), just ignore the file specific logic and return the directory path for each directory. added tests